### PR TITLE
Use sphinx-galery 0.9 and simplify conf.py

### PR DESCRIPTION
**Pull request by NicolasHug**
Now that sphinx-gallery 0.9 is released, we can simplify the logic of `inject_minigalleries()`.

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 sphinx==2.4.4
-sphinx-gallery
+sphinx-gallery>=0.9.0
 matplotlib
 numpy
```
Comment (oke-aditya) [inline docs/requirements.txt:4]: Actually, we can avoid NumPy dependency. `matplotlib` covers it?

Comment (NicolasHug) [reply docs/requirements.txt:4]: You're right, but it doesn't hurt to be explicit for those things IMHO.
Who knows: we might end up not using matplotlib for showing images, or mpl may even not depend on numpy at some point (quite unlikely I agree)

**Review by datumbox — APPROVED** [review]
LGTM

File: docs/requirements.txt (hunk 1 above)
Comment (datumbox) [reply docs/requirements.txt:4]: I think the best practice is to we avoid transitive dependencies and define explicitly. With a quick look on the documentation folder, I don't see a direct import on `numpy` (only some references on the config) but if it is a necessary dependency I would leave it explicitly defined for the reasons that Nicolas explained.

Comment (NicolasHug) [reply docs/requirements.txt:4]: We have an explicit dep on numpy in `gallery/plot_transforms.py` which isn't in the doc folder, but it's still part of the doc building

**Pull request merged**